packages/beacon-node/src/api/impl/beacon/blocks/index.ts

```diff
@@ -297,17 +299,22 @@ export function getBeaconBlockApi({
     const source = ProducedBlockSource.builder;
     chain.logger.debug("Reconstructing  signedBlockOrContents", {slot, blockRoot, source});
 
-    const signedBlockOrContents = await reconstructBuilderBlockOrContents(chain, {
+    const signedBlockOrContents = await reconstructBuilderBlockOrContents(fork, chain, {
       data: signedBlindedBlock,
       bytes: context?.sszBytes,
     });
 
-    // the full block is published by relay and it's possible that the block is already known to us
-    // by gossip
-    //
-    // see: https://github.com/ChainSafe/lodestar/issues/5404
-    chain.logger.info("Publishing assembled block", {slot, blockRoot, source});
-    return publishBlock({signedBlockOrContents}, {...context, sszBytes: null}, {...opts, ignoreIfKnown: true});
+    if (signedBlockOrContents !== undefined) {
+      // the full block is published by relay and it's possible that the block is already known to us
+      // by gossip
+      //
+      // see: https://github.com/ChainSafe/lodestar/issues/5404
+      chain.logger.info("Publishing assembled block", {slot, blockRoot, source});
+      return publishBlock({signedBlockOrContents}, {...context, sszBytes: null}, {...opts, ignoreIfKnown: true});
+    }
+
+    // Skip publishing. This is an expected behaviour post-fulu
+    chain.logger.info("Skip publishing. Builder published the block on our behalf.");
```

Comment:
please add `{slot, blockRoot, source}` to the log, consider to define and use `logCtx` for. both logs

Reply:
I don't really like this log, we should just log somewhere that we submitted the block to the builder for publishing